src/bootstrap/dist.rs

```diff
@@ -598,6 +636,11 @@ pub fn extended(build: &Build, stage: u32, target: &str) {
     let cargo_installer = dist.join(format!("cargo-{}-{}.tar.gz",
                                             build.package_vers(&cargo_vers),
                                             target));
+    let rls_installer = dist.join(format!("{}.tar.gz",
+                                          pkgname(build, "rls")));
```

Comment:
This needs to be "{}-{}.tar.gz", with a final argument of `target`. That will put the target triple in the tarball filename, which is needed for disambiguation.